### DISCUSSION TITLE
What is the proper way to chain a list (IEnumerable) of EiherAsync/Aff  ?

### DISCUSSION THREAD

**Discussion by guillaumeagile** [louthy/language-ext#1286] [Q&A]
I have a IEnumerable<Aff<T>>, and I would like to await them wall, like Task.WhenAll does.
I cannot achieve to write it in a FP way. 
Can someone help?

**Comment by louthy** [comment]
```c#
IEnumerable<Aff<string>> mx = ...;

// Runs each item in the sequence after the previous one has completed
Aff<IEnumerable<string>> r2 = mx.SequenceSerial();

// Runs a number of the items in the sequence concurrently (see below)
Aff<IEnumerable<string>> r1 = mx.SequenceParallel();
```
With `SequenceParallel` you can also specify the maximum number of tasks to run concurrently:
```c#
var r1 = mx.SequenceParallel(10);
```
The default value is: `SysInfo.DefaultAsyncSequenceParallelism` which is defined to be `Math.Max(1, ProcessorCount / 2)`.  That property is mutable, so you can set the default for your whole application.

**Reply by guillaumeagile** [reply]
Thanks.
You confirm that Aff is the best structure (upon TryAsync or EitherAsync) to run a bunch of async operations ?
